compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -365,12 +365,12 @@ OMR::X86::CodeGenerator::initialize(TR::Compilation *comp)
    self()->setSupportsArrayCmp();
    self()->setSupportsArrayCopy();
 
-   if (comp->getOption(TR_EnableX86AdvancedMemorySet))
-      self()->setSupportsArraySet();
-
    static char *disableArraySet = feGetEnv("TR_disableArraySet");
```

Comment:
I personally prefer keep the code as it using an env var for being consistent with other platforms as well as historical code.
I agree that an option is more flexible and useful; but I think it might be better to introduce the option in a separate changeset and update all platforms to check the flag.